bolt-sidecar/benches/score_cache.rs

```diff
@@ -0,0 +1,81 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::collections::HashMap;
+
+use bolt_sidecar::common::score_cache::ScoreCache;
+
+const GET_SCORE: isize = 1;
+const INSERT_SCORE: isize = 2;
+const UPDATE_SCORE: isize = 3;
+
+fn bench_scorecache_vs_hashmap(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ScoreCache vs HashMap");
+
+    let mut score_cache: ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, usize, usize> =
+        ScoreCache::with_max_len(10_000);
+    let mut hash_map: HashMap<usize, usize> = HashMap::new();
+
+    // Insert benchmark
+    group.bench_function("ScoreCache Insert", |b| {
+        b.iter(|| {
+            for i in 0..1000 {
```

Comment:
Consider varying sizes of operations (1000, 10k, 100k) to understand how perf varies

Reply:
Since I want to measure the raw overhead of modifying the score when doing get, insert, update operations I don't think there is a big advantage in testing different sizes. Also it slows down the benchmarking by really a lot. Just 1000 iterations takes a couple of mins on a M3 pro cpu